core/git_command.py

```diff
@@ -245,38 +254,40 @@ def _repo_path(self, throw_on_stderr=True):
         Return the absolute path to the git repo that contains the file that this
         view interacts with.  Like `file_path`, this can be overridden by setting
         the view's `git_savvy.repo_path` setting.
-        """
-        def invalid_repo():
-            if throw_on_stderr:
-                raise ValueError("Unable to determine Git repo path.")
-            return None
 
+        Do not raise error when throw_on_stderr is `False`. It is needed
+        in GsUpdateStatusBarCommand, otherwise, spurious popup will be shown.
+        """
         # The below condition will be true if run from a WindowCommand and false
         # from a TextCommand.
         view = self.window.active_view() if hasattr(self, "window") else self.view
+        window = view.window()
         repo_path = view.settings().get("git_savvy.repo_path")
 
         if not repo_path or not os.path.exists(repo_path):
-            working_dir = self.find_working_dir()
-            if not working_dir or not os.path.isdir(working_dir):
-                return invalid_repo()
-
-            stdout = self.git(
-                "rev-parse",
-                "--show-toplevel",
-                working_dir=working_dir,
-                throw_on_stderr=throw_on_stderr
-                )
+            file_name = view.file_name()
 
-            repo_path = stdout.strip()
+            # try the current file first
+            if file_name:
+                repo_path = self.find_git_toplevel(
+                    os.path.dirname(file_name), throw_on_stderr=False)
+                # only set `git_savvy.repo_path` for current file
+                if repo_path:
+                    view.settings().set("git_savvy.repo_path", os.path.realpath(repo_path))
 
+            # fallback: use the first folder if the current file is not inside a git repo
             if not repo_path:
-                return invalid_repo()
+                if window:
+                    folders = window.folders()
+                    if folders:
+                        repo_path = self.find_git_toplevel(
+                            folders[0], throw_on_stderr=throw_on_stderr)
 
-            if view.file_name() and working_dir in view.file_name():
-                # only set repo_path for file in the working directory
-                # it allows a file living outside the repo to be edited in the current window
-                view.settings().set("git_savvy.repo_path", os.path.realpath(repo_path))
+            if not repo_path:
+                if throw_on_stderr:
+                    raise ValueError("Unable to determine Git repo path.")
```

Comment:
if the folder is not a git repo, only GsStatusBarEventListener will trigger and the `throw_on_stderr` argument will make sure it won't do anything.

If a user run a git command, e.g. `git: log`, it will report errors. Then,
- if there is an opened folder, pop up the initialization confirmation
- if there is no folder in the sidebar, it will just show a panel of `Unable to determine Git repo path`.